python/cudf/cudf/pylibcudf_tests/test_binaryops.py

```diff
@@ -0,0 +1,741 @@
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_EQUALS,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_max_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_MAX,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_min_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_MIN,
+    )
+
+
+@pytest.fixture(scope="module")
+def generic_binary_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.GENERIC_BINARY,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_logical_and_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_LOGICAL_AND,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_logical_or_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_LOGICAL_OR,
+    )
+
+
+@pytest.fixture(scope="module")
+def invalid_binary_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.INVALID_BINARY,
+    )
+
+
+def test_add(add_tests, columns):
+    _test_binaryop_inner(
+        add_tests, columns, np.add, plc.binaryop.BinaryOperator.ADD
+    )
+
+
+def test_sub(sub_tests, columns):
+    _test_binaryop_inner(
+        sub_tests, columns, np.subtract, plc.binaryop.BinaryOperator.SUB
+    )
+
+
+def test_mul(mul_tests, columns):
+    _test_binaryop_inner(
+        mul_tests, columns, np.multiply, plc.binaryop.BinaryOperator.MUL
+    )
+
+
+def test_div(div_tests, columns):
+    _test_binaryop_inner(
+        div_tests, columns, np.divide, plc.binaryop.BinaryOperator.DIV
+    )
+
+
+def test_true_div(true_div_tests, columns):
+    _test_binaryop_inner(
+        true_div_tests,
+        columns,
+        np.true_divide,
+        plc.binaryop.BinaryOperator.TRUE_DIV,
+    )
+
+
+def test_floor_div(floor_div_tests, columns):
+    with warnings.catch_warnings(record=True):
+        warnings.filterwarnings("ignore", message="divide by zero")
+        _test_binaryop_inner(
+            floor_div_tests,
+            columns,
+            np.floor_divide,
+            plc.binaryop.BinaryOperator.FLOOR_DIV,
+        )
+
+
+def test_mod(mod_tests, columns):
+    _test_binaryop_inner(
+        mod_tests, columns, np.mod, plc.binaryop.BinaryOperator.MOD
+    )
+
+
+def test_pmod(pmod_tests, columns):
+    with warnings.catch_warnings(record=True):
```

Comment:
[Prefer pytest.warns to warnings.catch_warnings](https://docs.rapids.ai/api/cudf/stable/developer_guide/testing/#testing-code-that-throws-warnings)